src/Uno.Foundation/HResult.cs

```diff
@@ -1,12 +1,17 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Text;
 
 namespace Windows.Foundation
 {
-	public struct HResult
+	public struct HResult : IEquatable<HResult>
 	{
 		/// <summary>An integer that describes an error.</summary>
 		public int Value;
+
+		public override bool Equals(object obj) => obj is HResult result && Equals(result);
+		public bool Equals(HResult other) => Value == other.Value;
+		public override int GetHashCode() => -1937169414 + Value.GetHashCode();
+
+		public static bool operator ==(HResult left, HResult right) => left.Equals(right);
+		public static bool operator !=(HResult left, HResult right) => !(left == right);
```

Comment:
minor but, for the same reason (trimmability), it would be better to have `!=` call `!Equals` (that can't be trimmed) so the linker can eliminate `==` if unused.